Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java

```diff
@@ -1431,4 +1431,27 @@ static void loadPartnerParams(JSONObject body, BranchPartnerParameters partnerPa
         }
         body.put(Defines.Jsonkey.PartnerData.getKey(), partnerData);
     }
+
+    /**
+     * <p>Gets the {@link Defines.BranchAttributionLevel} value that has been set.</p>
+     *
+     * @return A {@link Defines.BranchAttributionLevel} value representing the attribution level set.
+     */
+    public Defines.BranchAttributionLevel getConsumerProtectionAttributionLevel() {
+        int levelInt = getInteger(KEY_CONSUMER_PROTECTION_ATTRIBUTION_LEVEL, -1);
+        if (levelInt == -1) {
+            return null;
+        } else {
+            return Defines.BranchAttributionLevel.values()[levelInt];
+        }
+    }
+
+    /**
+     * <p>Sets the {@link Defines.BranchAttributionLevel} value via the Branch API.</p>
+     *
+     * @param preference A {@link Defines.BranchAttributionLevel} value containing the desired attribution level.
+     */
+    public void setConsumerProtectionAttributionLevel(Defines.BranchAttributionLevel preference) {
+        setInteger(KEY_CONSUMER_PROTECTION_ATTRIBUTION_LEVEL, preference.ordinal());
```

Comment:
Is there a reason why we're using integer representation instead of string for logic? Seems prone to error if the enum gets updated.